src/scaladoc/scala/tools/nsc/doc/model/ModelFactory.scala

```diff
@@ -909,13 +909,12 @@ class ModelFactory(val global: Global, val settings: doc.Settings) {
   def makeTypeInTemplateContext(aType: Type, inTpl: TemplateImpl, dclSym: Symbol): TypeEntity = {
     def ownerTpl(sym: Symbol): Symbol =
       if (sym.isClass || sym.isModule || sym == NoSymbol) sym else ownerTpl(sym.owner)
-    val tpe =
-      if (thisFactory.settings.useStupidTypes) aType else {
-        def ownerTpl(sym: Symbol): Symbol =
-          if (sym.isClass || sym.isModule || sym == NoSymbol) sym else ownerTpl(sym.owner)
-        val fixedSym = if (inTpl.sym.isModule) inTpl.sym.moduleClass else inTpl.sym
-        aType.asSeenFrom(fixedSym.thisType, ownerTpl(dclSym))
-      }
+    val tpe = {
+      def ownerTpl(sym: Symbol): Symbol =
```

Comment:
copy and paste?
isn't this the same as the shadowed ownerTpl on line 910

Reply:
lol, indeed. I've pushed a revised commit that keeps only one of the two copies.